raiden-ts/src/transfers/epics/locked.ts

```diff
@@ -477,6 +492,304 @@ function makeAndSignWithdrawConfirmation(
+      assert(transfer.chain_id.eq(network.chainId), 'chainId mismatch');
+      assert(transfer.channel_identifier.eq(channel.id), 'channelId mismatch');
+      assert(transfer.nonce.eq(nextNonce(channel.partner.balanceProof)), 'nonce mismatch');
+      assert(
+        transfer.transferred_amount.eq(channel.partner.balanceProof?.transferredAmount ?? Zero),
+        'transferredAmount mismatch',
+      );
+      assert(
+        transfer.locked_amount.eq(
+          (channel.partner.balanceProof?.lockedAmount ?? Zero).add(transfer.lock.amount),
+        ),
+        'lockedAmount mismatch',
+      );
+
+      const partnerCapacity = channelAmounts(channel).partnerCapacity;
+      assert(
+        transfer.lock.amount.lte(partnerCapacity),
+        'balanceProof total amount bigger than capacity',
+      );
+
+      assert(transfer.recipient === address, "Received transfer isn't for us");
+
+      assert(
+        transfer.lock.expiration.sub(state.blockNumber).gt(revealTimeout),
+        'lock expires too soon',
+      );
+      const locks = [...(channel.partner.locks ?? []), transfer.lock];
+      const locksroot = getLocksroot(locks);
+      assert(transfer.locksroot === locksroot, 'locksroot mismatch');
+      const token = findKey(state.tokens, tn => tn === tokenNetwork)! as Address;
+
+      log.info(
+        'Receiving transfer of value',
+        transfer.lock.amount.toString(),
+        'of token',
+        token,
+        ', from',
+        transfer.initiator,
+        ', through partner',
+        partner,
+      );
+
+      let request$: Observable<Signed<SecretRequest> | undefined> = of(undefined);
+      if (!caps?.[Capabilities.NO_RECEIVE] && transfer.target === address)
+        request$ = defer(() => {
+          const request: SecretRequest = {
+            type: MessageType.SECRET_REQUEST,
+            payment_identifier: transfer.payment_identifier,
+            secrethash,
+            amount: transfer.lock.amount,
+            expiration: transfer.lock.expiration,
+            message_identifier: makeMessageId(),
+          };
+          return signMessage(signer, request, { log });
+        });
+
+      const processed$ = defer(() => {
+        const processed: Processed = {
+          type: MessageType.PROCESSED,
+          // eslint-disable-next-line @typescript-eslint/camelcase
+          message_identifier: transfer.message_identifier,
+        };
+        return signMessage(signer, processed, { log });
+      });
+
+      // if any of these signature prompts fail, none of these actions will be emitted
+      return combineLatest([processed$, request$]).pipe(
+        mergeMap(function*([processed, request]) {
+          yield transferSigned({ message: transfer, fee: Zero as Int<32> }, meta);
+          // request initiator's presence, to be able to request secret
+          yield matrixPresence.request(undefined, { address: transfer.initiator });
```

Comment:
Do we have to request the presence, even though we may not request the secret?

Reply:
Nice catch! You're right, if `NO_RECEIVE`  is truty, we don't need to whitelist/request presence from initiator.